### PR TITLE
nc_news_8

### PR DESCRIPTION
**Pull request by theRohirrim**
completes functionality and tests for PATCH /api/articles/:article_id. Updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -296,3 +296,44 @@ describe("POST: /api/articles/:article_id/comments", () => {
         .expect(404)
     });
 })
+
+describe("PATCH /api/articles/:article_id", () => {
+    test("200: Successful patch will return the article", () => {
```
Comment (saimakhanom) [inline __tests__/app.test.js:301]: We should have a test that checks whether the request is also able to decrement the votes on the article

File: __tests__/app.test.js [hunk 2]
```diff
@@ -296,3 +296,44 @@ describe("POST: /api/articles/:article_id/comments", () => {
...
+        })
+    })
+
+    test("400: Rejects based on a bad request (article_id)", () => {
+        return request(app)
+        .patch("/api/articles/not_article")
+        .send({
+            inc_votes: 5
+        })
+        .expect(400)
```
Comment (saimakhanom) [inline __tests__/app.test.js:328]: we should also have a message to accompany this error code

File: __tests__/app.test.js [hunk 3]
```diff
@@ -296,3 +296,44 @@ describe("POST: /api/articles/:article_id/comments", () => {
...
+    });
+
+    test("404: Rejects based on a valid but non existing article_id)", () => {
+        return request(app)
+        .patch("/api/articles/5555")
+        .send({
+            inc_votes: 5
+        })
+        .expect(404)
```
Comment (saimakhanom) [inline __tests__/app.test.js:337]: we should also have a message to accompany this error code

**Comment by saimakhanom** [comment]
To be able to merge, you need at least the test for a bad request where there is an incomplete request body, and the messages accompanying the 400, 404 responses. The others tests are a bonus :)

**Review by MKD01 — APPROVED** [review]
Good work, im happy for this to be merged 👍

**Pull request merged**